include/RAJA/policy/openmp/forall.hpp

```diff
@@ -66,12 +67,15 @@ RAJA_INLINE void forall_impl(const omp_parallel_exec<InnerPolicy>&,
                              Iterable&& iter,
                              Func&& loop_body)
 {
-#pragma omp parallel
-  {
-    using RAJA::internal::thread_privatize;
-    auto body = thread_privatize(loop_body);
-    forall_impl(InnerPolicy{}, std::forward<Iterable>(iter), body.get_priv());
-  }
+    
+  RAJA::Region<RAJA::omp_parallel_region>([=](){
+
+      using RAJA::internal::thread_privatize;
+      auto body = thread_privatize(loop_body);
+      forall_impl(InnerPolicy{}, iter, body.get_priv());
+
```

Comment:
@trws , removing std::forward didn't quite do the trick. The code didn't build.

Reply:
Ok, what's the error?

Reply:
I got the following: 
/g/g17/vargas45/Git-Repo/RAJA/region/include/RAJA/pattern/detail/forall.hpp:23:39: error: no matching function for call to 'begin(const RAJA::nested::IterableTiler<const RAJA::TypedRangeStrideSegment<long int>&>&)'
   auto begin##SUFFIX = begin(CONTAINER);             \
                                       ^
/g/g17/vargas45/Git-Repo/RAJA/region/include/RAJA/pattern/detail/forall.hpp:27:40: note: in expansion of macro 'RAJA_EXTRACT_BED_SUFFIXED'
 #define RAJA_EXTRACT_BED_IT(CONTAINER) RAJA_EXTRACT_BED_SUFFIXED(CONTAINER, _it)
                                        ^
/g/g17/vargas45/Git-Repo/RAJA/region/include/RAJA/policy/openmp/forall.hpp:106:3: note: in expansion of macro 'RAJA_EXTRACT_BED_IT'
   RAJA_EXTRACT_BED_IT(iter);
   ^
/g/g17/vargas45/Git-Repo/RAJA/region/include/RAJA/pattern/detail/forall.hpp:23:39: note: candidates are:
   auto begin##SUFFIX = begin(CONTAINER);             \
                                       ^
/g/g17/vargas45/Git-Repo/RAJA/region/include/RAJA/pattern/detail/forall.hpp:27:40: note: in expansion of macro 'RAJA_EXTRACT_BED_SUFFIXED'
 #define RAJA_EXTRACT_BED_IT(CONTAINER) RAJA_EXTRACT_BED_SUFFIXED(CONTAINER, _it)
                                        ^
/g/g17/vargas45/Git-Repo/RAJA/region/include/RAJA/policy/openmp/forall.hpp:106:3: note: in expansion of macro 'RAJA_EXTRACT_BED_IT'
   RAJA_EXTRACT_BED_IT(iter);

Reply:
The relevant part of the error is that begin, and end, discard qualifiers.  `IterableTiler::begin` and end weren't marked const, even though they don't modify anything.